src/firmware/shared/math/matrix.h

```diff
@@ -1,73 +1,137 @@
+#pragma once
 
-
+#include <stdlib.h>
 /**
  * A representation of a matrix that has an arbitrary number of rows and columns specified
  * by n_cols and n_rows. The recommended way to use this struct is to use the get_matrix
  * function, which will allocate the memory required for each of the rows in the matrix
  * and each of the entries in each of those rows. It will also assign the n_cols and
- * n_rows for the matrix. Then when you are done with the matrix, you will need to call
- * free_matrix.
+ * n_rows for the matrix.
+ *
+ * *******************************************************************
+ * NOTE: Then when you are done with the matrix, you will need to call
+ * shared_matrix_destroy(matrix).
+ * ********************************************************************
  */
-typedef struct
-{
-    float **rows;
-    int n_cols;
-    int n_rows;
-} Matrix;
+typedef struct Matrix Matrix_t;
 
 /**
- * Gets a matrix of the given size. This uses dynamic memory allocation so make sure to
- * call free_matrix when you are done using it if you don't want memory leaks.
+ * Creates a Matrix with the given numbers of rows and columns
  *
- * @param n_rows the number of rows in the matrix
- * @param n_cols the number of columns in the matrix
- * @return a Matrix object
+ * @param n_rows [in] The number of rows in the matrix
+ * @param n_cols [in] The number of columns in the matrix
+ *
+ * @return Pointer to the created Matrix
  */
-Matrix create_matrix(int n_rows, int n_cols);
+Matrix_t* shared_matrix_create(unsigned int n_rows, unsigned int n_cols);
 
 /**
  * Frees the memory used up by a matrix.
  *
- * @param matrix the matrix to free
+ * @param matrix [in] The matrix to destroy
  */
-void free_matrix(Matrix matrix);
+void shared_matrix_destroy(Matrix_t* matrix);
 
 /**
  * Multiplies two matrices together and returns the resulting matrix, which is of size
  * A.n_rows x B.n_cols
  *
- * @param A the left matrix
- * @param B the right matrix
+ * @param A [in] the left matrix
```

Comment:
nit: @param capitalization should be consistent 
```suggestion
 * @param A [in] The left matrix
```